File: app/src/main/java/com/glsebastiany/bakingapp/injection/module/ApplicationModule.java
package com.glsebastiany.bakingapp.injection.module;

import android.content.Context;

import com.glsebastiany.bakingapp.MyApplication;
import com.glsebastiany.bakingapp.injection.scopes.ApplicationContext;
import com.glsebastiany.bakingapp.repository.RecipesRepository;

import dagger.Module;
import dagger.Provides;


@Module
public class ApplicationModule {

    private MyApplication application;

    public ApplicationModule(MyApplication application) {
        this.application = application;
    }

    @Provides
    @ApplicationContext
    public MyApplication providesApplication() {
        return application;
    }

    @Provides
    @ApplicationContext
    public Context provideContext() {
        return application;
    }

    @Provides
    @ApplicationContext
    public RecipesRepository provideRecipeRepository() {
        return new RecipesRepository(application);
    }

}

File: app/src/main/java/com/glsebastiany/bakingapp/injection/module/ApplicationModule.kt
package com.glsebastiany.bakingapp.injection.module

import android.content.Context
import com.glsebastiany.bakingapp.MyApplication
import com.glsebastiany.bakingapp.injection.scopes.ApplicationContext
import com.glsebastiany.bakingapp.repository.RecipesRepository
import dagger.Module
import dagger.Provides


@Module
open class ApplicationModule(private val application: MyApplication) {

    @Provides
    @ApplicationContext
    fun providesApplication(): MyApplication = application

    @Provides
    @ApplicationContext
    fun provideContext(): Context = application

    @Provides
    @ApplicationContext
    open fun provideRecipeRepository(): RecipesRepository = RecipesRepository(application)

}
